packages/server/src/utilities/rowProcessor/utils.ts

```diff
@@ -40,52 +45,50 @@ export function fixAutoColumnSubType(
 /**
  * Looks through the rows provided and finds formulas - which it then processes.
  */
-export function processFormulas(
+export function processFormulas<T extends Row | Row[]>(
```

Comment:
What's the associativity here? Is this saying `T extends Row | T extends Row[]`? Or is it `(T extends Row) | Row[]`?

Reply:
it works like `T extends Row | T extends Row[]` - we've used a format like this before a few times, for something like this where a function can process either type, its nice to be able to make sure it denotes the correct response type as well, so this function is aware that if type `Row` is passed in, you'll get `Row` out, or vice versa for `Row[]`